### PR TITLE
fix(msvc.build.warning.C4702): unreachable code

### PR DESCRIPTION
**Pull request by stef145g**

### CODE REVIEW

File: src/db/DB_Table_Subcategory_V1.h [hunk 1]
```diff
@@ -100,60 +99,6 @@ struct DB_Table_SUBCATEGORY_V1 : public DB_Table
         return true;
     }
 
-    bool ensure_data(wxSQLite3Database* db)
-    {
-        try
-        {
-            db->ExecuteUpdate(wxString::Format("INSERT INTO SUBCATEGORY_V1 VALUES (1, '%s', 1)", wxTRANSLATE("Telephone")));
```
Comment (vomikan) [inline src/db/DB_Table_Subcategory_V1.h]: Should be
db->ExecuteUpdate(wxString::Format("INSERT INTO SUBCATEGORY_V1 VALUES (1, '%s', 1)", _("Telephone")));

File: src/db/DB_Table_Subcategory_V1.h [hunk 2]
```diff
@@ -100,60 +99,6 @@ struct DB_Table_SUBCATEGORY_V1 : public DB_Table
...
-            db->ExecuteUpdate(wxString::Format("INSERT INTO SUBCATEGORY_V1 VALUES (2, '%s', 1)", wxTRANSLATE("Electricity")));
```
Comment (vomikan) [inline src/db/DB_Table_Subcategory_V1.h]: May be better to init two integer variables like c=1, s=1 nad use it
 db->ExecuteUpdate(wxString::Format("INSERT INTO SUBCATEGORY_V1 VALUES (++c, '%s', s)", wxTRANSLATE("Electricity")));

File: util/sqlite2cpp.py [hunk 3]
```diff
@@ -178,26 +177,8 @@ def to_string(self, sql = None):
...
     
-        s += '''
-    bool ensure_data(wxSQLite3Database* db)
-    {
-        try
-        {'''
 
-        for r in self._data:
-            s += '''
-            db->ExecuteUpdate(wxString::Format("INSERT INTO %s VALUES (%s)", %s));''' % (self._table, ', '.join(["'%s'" if isinstance(i, unicode) else str(i) for i in r]), ', '.join([ 'wxTRANSLATE("' + i + '")' for i in r if isinstance(i, unicode)]))
```
Comment (vomikan) [inline util/sqlite2cpp.py]: Please remove wxTRANSLATE from this line.

**Comment by stef145g** [comment]
New commit reverts the previous code and adds extra code. Also fixed the translation issue from a previous update

**Comment by guanlisheng** [comment]
i don't think we such dummy code here and am wondering why you guys have so strong option to remove a waring...

i think the translation is incorrect while wxTRANSLATE 
http://docs.wxwidgets.org/3.0/group__group__funcmacro__string.html#gab8862de0f74049708bfea75ae9e14c45

**Comment by guanlisheng** [comment]
_() is not correct here

http://docs.wxwidgets.org/3.0/group__group__funcmacro__string.html#ga8a02b8875a521df57263a9e6f090f2d0

**Comment by stef145g** [comment]
@guanlisheng I have tested the category translations by creating a new database using Italian as my base language.
Using the wxTRANSLATE ("string")  - the categories in the database end up in English.
Using the _("string")  - the categories in the database end up in Italian.
 
It appears that @vomikan is correct and _("string") should be used here.

**Comment by guanlisheng** [comment]
The desired behavior is to store **Englis** in the database.

**Comment by stef145g** [comment]
This means that the current implementation needs to be fixed to show the correct language for categories. Now the current implementation will only show English.

**Comment by vomikan** [comment]
In that case you can'can't get translated categories by sql. And any new
user created categories will be in national language.

суббота, 13 февраля 2016 г. пользователь Lisheng написал:

**Comment by gabriele-v** [comment]
Completely agree with @stef145g and @vomikan : 
Why we need to store categories in English and add more code to translate them?
Categories are only labels, so better to show them directly as written in DB.
If new category created how does it work with translation and DB save? It's a no-sense..

Regarding build warning, you said lots of times to improve coding and then you leave a warning that could be fixed? I don't understand..
Also, current implementation of ensure_data it's useless because if it doesn't execute any statement in try, what error will be catched?

**Comment by guanlisheng** [comment]
@vomikan @stef145g @gabriele-v 
Regarding English or translated strings, are you guys kidding for such simple question? do we know why we store the ISO date format (YYYY-MM-DD) in database? if negative,  considering the simplest case that final users switch the language setting from **Italian** to **Russian**, how to deal with the existing records in table CATEGORY_V1, SUBCATEGORY_V1 and CURRENCYFORMATS_V1? as what i said times, please distinguish MVC during the software developing. to this particular case, all transaction should be done during the displaying (V in the MVC). 

About the building warnings and code quality/improvement, the empty try statemenet is not perfect, even problematic while we should put more time on how we design, use the c++ class and initialize its members. 
Leave current warnings there and the coming demo data will fill with the empty try statement.

**Comment by gabriele-v** [comment]
I could ask you the same about kidding, you can't absolutely compare a date with a string..
Date it's stored in ISO date only because SQLite doesn't have a proper date field. If MMEX will be on SQL Server or MySQL with proper date type field, problem won't exist at all.

Switching language has only to switch GUI interface translation (and fixed enum like void, reconciled etc..), not data!
I didn't know a software that will switch a user-customizable label or data when you change language. It's like asking to Gmail or Outlook to change categories\tags when you change GUI language..
Completely no-sense up to me.

Consider the case that a user adds a new category outside of ones provided, in which language you will add it to database?

**Comment by vomikan** [comment]
I am on the same point as Gabriele.

воскресенье, 14 февраля 2016 г. пользователь Gabriele-V написал:

**Comment by gabriele-v** [comment]
@guanlisheng , I think that there was a misunderstanding here.

I was referring to "normal" user data, not default ones.
I could agree with you regarding "default" data, for example in Outlook.com default categories are translated if you change language: practically it would mean that, when viewing\showing a category, we have to search if there is a translation for that category, otherwise display as saved, correctly?

This approach works, but IMHO it will add some complexity to MMEX that we currently don't need, in particular with GRM and SQL..

**Comment by gabriele-v** [comment]
Consider also case when a category translation is changed in a new MMEX version or if a translation it's different from Desktop \ Android \ WebApp \ iOS (because translators are different people), users will be really disoriented..

This to obtain what? Only simpler language change after DB initialization?
I don't think that lots of users will change language after initialization.
Technically speaking I could agree with you but IMHO here it isn't worth the trouble..

**Comment by vomikan** [comment]
There are only two 'standard' categories  in mmex:
Transfer and Unknown
All others may be redefine by user.

воскресенье, 14 февраля 2016 г. пользователь Gabriele-V написал:

**Comment by guanlisheng** [comment]
I think everything is pretty clear now since we were/are talking the default/predefined data instead of the regular user data which input from the GUI instead of the raw SQL here.

Regarding the value, i think it's a pretty interesting topic and i would like use the system language to disable the language switch (via #340).  Even this,  we have to have fully examine on whether current code works since i18n is variables.


NOTE, i don't change any existing behavior and what i did is to move the DB initialization from manual written cpp code to auto generated one....  you guys' response is really weird.

**Comment by gabriele-v** [comment]
I don't know why you continue to close this pull request.

This it's not true:

> NOTE, i don't change any existing behavior and what i did is to move the DB initialization from manual written cpp code to auto generated one.... you guys' response is really weird.

because old code will initialize (and show) categories with user language, now they are always initialized (and showed) in english.

So at least until you have completed the automatic language translation IMHO it should be reverted as before.

**Comment by guanlisheng** [comment]
you are wasting my time for such stupid discussion and please read the code firstly before putting your statement!


here are the link to the old code before my commit.
https://github.com/moneymanagerex/moneymanagerex/blob/v1.2.5/src/model/Model_Currency.cpp
https://github.com/moneymanagerex/moneymanagerex/blob/v1.2.5/src/model/Model_Category.cpp

**Comment by gabriele-v** [comment]
1.2.5 => Italian language, creating new DB categories inserted in italian
master @https://github.com/moneymanagerex/moneymanagerex/commit/53ff8912bef5d0b3437f305290a4a43873651352 => Italian language, creating new DB categories inserted in italian
master @https://github.com/moneymanagerex/moneymanagerex/commit/2cfc7feafabc120f74fd9560f7f59cd168836db4 => Italian language, creating new DB categories inserted in english

You are wasting my time to make tests and please test your code after changes!

**Comment by guanlisheng** [comment]
Then you can switch your language setting back to English to see what happened....

i did not notice the special code for category/sub category which is not consist w/ other ones.

Anyway, it's not acceptable to store translated strings in database tables.

**Comment by stef145g** [comment]
A new pull request #712 has been opened to correct the warning only. Improved version.

**Pull request closed without merging**